awx/main/tasks.py

```diff
@@ -409,22 +409,26 @@ def cleanup_execution_environment_images():
 
 
 @task(queue=get_local_queuename)
-def execution_node_health_check(node):
+def node_remote_health_check(node):
+    if node == '':
+        logger.warn('Remote health check incorrectly called with blank string')
+        return
     try:
         instance = Instance.objects.get(hostname=node)
     except Instance.DoesNotExist:
         logger.warn(f'Instance record for {node} missing, could not check capacity.')
         return
-    data = worker_info(node)
+
+    data = worker_info(node, work_type='ansible-runner' if instance.node_type == 'execution' else 'local')
 
     prior_capacity = instance.capacity
 
     instance.save_health_data(
-        version='ansible-runner-' + data.get('runner_version', '???'),
+        version=('ansible-runner-' + data.get('runner_version', '???')) if instance.node_type == 'execution' else None,
         cpu=data.get('cpu_count', 0),
         memory=data.get('mem_in_bytes', 0),
-        uuid=data.get('uuid'),
-        has_error=bool(data.get('errors')),
+        uuid=data.get('uuid') if instance.node_type == 'execution' else None,  # TODO, have --worker-info read from settings file
+        errors='\n'.join(data.get('errors', [])),
```

Comment:
This just keeps getting uglier, and I'm leaning towards redesigning this to use the dispatcher system as opposed to the receptor mesh for main cluster nodes.

Reply:
I actually really like this idea, the dispatcher system is where I would expect this to be rather than over the mesh.

Reply:
Yeah. The output doesn't work for me on dev env:

```
{
    "uuid": "00000000-0000-0000-0000-000000000000",
    "hostname": "awx_1",
    "version": "19.2.2",
    "last_health_check": "2021-09-03T14:37:08.587927Z",
    "errors": "",
    "cpu": 8,
    "memory": 6030827520,
    "cpu_capacity": 32,
    "mem_capacity": 37,
    "capacity": 37
}
```

Reply:
@thenets that output is perfectly fine. What did you think is wrong with it?

Reply:
@AlanCoding I thought the `uuid` would be something different than zeros. If this was expected, perfect :)

Reply:
right, the dev environment just never set that up. Someone could do it, but it's not specific to the current feature work.